src/cpu/flexcpu/FlexCPU.py

```diff
@@ -0,0 +1,91 @@
+    # otherwise happen immediately to the earliest tick on a clock edge (May be
+    # the same tick if already on a clock edge).
+    clocked_dtb_translation = Param.Bool(Self.clocked_cpu, "Ties requests for "
+                                         "data addresses to clock edges.")
+    clocked_inst_fetch = Param.Bool(Self.clocked_cpu, "Ties requests for "
+                                    "instruction data to clock edges.")
+    clocked_itb_translation = Param.Bool(Self.clocked_cpu, "Ties requests for "
+                                         "instruction addresses to "
+                                         "clock edges.")
+    clocked_memory_request = Param.Bool(Self.clocked_cpu, "Ties requests for "
+                                        "sending packets to memory to clock "
+                                        "edges")
+
+    clocked_execution = Param.Bool(Self.clocked_cpu, "Ties requests for "
+                                   "instruction execution to clock edges.")
+    execution_latency = Param.Cycles(1, "Number of cycles for each "
+                                         "instruction to execute")
+    execution_bandwidth = Param.Int(0, "Number of executions each cycle. "
+                                       "Assumes a fully-pipelined unit and 0 "
+                                       "implies infinite bandwidth.")
+
+    fetch_buffer_size = Param.Unsigned(Parent.cache_line_size,
+                                       "Size of fetch buffer in bytes. "
+                                       "Also determines size of fetch "
+                                       "requests. Should not be larger than a "
+                                       "cache line.")
+
+    instruction_buffer_size = Param.Unsigned(0, "Size of the dynamic "
+                                             "instruction buffer. This buffer "
+                                             "is used for maintaining the "
+                                             "commit order of instructions. "
+                                             "Limiting this limits the number "
+                                             "of instructions that can be "
+                                             "handled at once before commit "
+                                             "(out of order). 0 implies an "
+                                             "infinitely large buffer.")
+
+    clocked_issue = Param.Bool(Self.clocked_cpu, "Ties requests for issuing "
+                                                 "instructions to the clock")
+    issue_latency = Param.Cycles(0, "Number of cycles each instruction takes "
+                                    "to issue.")
+    issue_bandwidth = Param.Int(0, "Number of instructions/micro-ops that can "
+                                   "be issued each cycle.")
+
+
+    strict_serialization = Param.Bool(True, "Controls behavior of serializing "
```

Comment:
What does `True` imply? What is "strict" serialization?

Reply:
expanded the comment